File: extensions/aether-forge/kotlin/src/main/kotlin/forge/ForgeEntry.kt
package forge

import java.time.Instant
import java.util.UUID

data class ForgeEntry(
    val id: UUID = UUID.randomUUID(),
    val packageId: String,
    val ecosystem: String,
    val version: String,
    val name: String,
    val description: String = "",
    val author: String = "",
    val licenseId: String = "",
    val sizeBytes: Long = 0L,
    val checksum: String,
    val checksumAlgorithm: String = "sha256",
    val downloadUrl: String,
    val mirrorUrls: List<String> = emptyList(),
    val dependencies: List<String> = emptyList(),
    val tags: List<String> = emptyList(),
    val isVerified: Boolean = false,
    val downloadCount: Long = 0L,
    val cachedAt: Instant = Instant.now(),
    val expiresAt: Instant? = null,
    val metadata: Map<String, String> = emptyMap()
)

File: extensions/aether-forge/swift/Sources/AetherNetForge/ForgeEntry.swift
import Foundation

public struct ForgeEntry: Codable, Equatable {
    public let id: UUID
    public let packageId: String
    public let ecosystem: String
    public let version: String
    public let name: String
    public var description: String
    public var author: String
    public var licenseId: String
    public var sizeBytes: Int64
    public let checksum: String
    public var checksumAlgorithm: String
    public let downloadUrl: String
    public var mirrorUrls: [String]
    public var dependencies: [String]
    public var tags: [String]
    public var isVerified: Bool
    public var downloadCount: Int64
    public let cachedAt: Date
    public var expiresAt: Date?
    public var metadata: [String: String]

    public init(
        id: UUID = UUID(),
        packageId: String,
        ecosystem: String,
        version: String,
        name: String,
        description: String = "",
        author: String = "",
        licenseId: String = "",
        sizeBytes: Int64 = 0,
        checksum: String,
        checksumAlgorithm: String = "sha256",
        downloadUrl: String,
        mirrorUrls: [String] = [],
        dependencies: [String] = [],
        tags: [String] = [],
        isVerified: Bool = false,
        downloadCount: Int64 = 0,
        cachedAt: Date = Date(),
        expiresAt: Date? = nil,
        metadata: [String: String] = [:]
    ) {
        self.id = id
        self.packageId = packageId
        self.ecosystem = ecosystem
        self.version = version
        self.name = name
        self.description = description
        self.author = author
        self.licenseId = licenseId
        self.sizeBytes = sizeBytes
        self.checksum = checksum
        self.checksumAlgorithm = checksumAlgorithm
        self.downloadUrl = downloadUrl
        self.mirrorUrls = mirrorUrls
        self.dependencies = dependencies
        self.tags = tags
        self.isVerified = isVerified
        self.downloadCount = downloadCount
        self.cachedAt = cachedAt
        self.expiresAt = expiresAt
        self.metadata = metadata
    }
}
